lib/bn.js

```diff
@@ -1021,6 +1237,12 @@ BN.prototype.mul = function mul(num) {
   return this.mulTo(num, out);
 };
 
+// Multiply employing FFT
+BN.prototype.mulf = function mulf(num) {
+  var out = new BN(null);
+  return jumboMulTo(this, num, out);
```

Comment:
It's not about changing the design -- i'm always open to change things the good way -- it's about messing the design: i see no graceful way to apply your suggestions. And performance-wise pros aren't clear for me to sacrifice readability & changeability of the code.